react/features/base/dialog/components/MessageDialog.web.js

```diff
@@ -0,0 +1,170 @@
+         * of the dialog.
+         */
+        message: React.PropTypes.string,
+
+        /**
+         * Invoked to obtain translated strings.
+         */
+        t: React.PropTypes.func
+    };
+
+    /**
+     * Initializes a new {@code MessageDialog} instance.
+     *
+     * @param {Object} props - The read-only properties with which the new
+     * instance is to be initialized.
+     */
+    constructor(props) {
+        super(props);
+
+        // Bind event handlers so they are only bound once for every instance.
+        this._onDontShowChanged = this._onDontShowChanged.bind(this);
+        this._onSubmit = this._onSubmit.bind(this);
+    }
+
+    /**
+     * Load from local storage the option does user had indicated that we should
+     * not show this message dialog.
+     *
+     * @inheritdoc
+     * @returns {void}
+     */
+    componentWillMount() {
+        if (this.props.doNotShowWarningLocalStorageKey) {
+            this.setState({
+                doNotShowDialog:
+                    jitsiLocalStorage.getItem(
+                        this.props.doNotShowWarningLocalStorageKey)
+                            === 'true'
+            });
+        } else {
+            this.setState({
+                doNotShowDialog: false
+            });
+        }
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        if (this.state.doNotShowDialog) {
+            return null;
+        }
+
+        const doNotShowWarning = this._renderDoNotShowWarning();
+
+        return (
+            <Dialog
+                onSubmit = { this._onSubmit }
+                { ...this.props }>
+                { this.props.message }
+                { this.props.children }
+                { doNotShowWarning }
+            </Dialog>
+        );
+    }
+
+    /**
+     * Creates a 'do not show dialog' component that contains a toggle which
+     * user can use to indicate that this is the last time he sees this dialog.
+     * If 'do not show dialog' component is not enabled returns null.
+     *
+     * @private
+     * @returns {ReactComponent|null}
+     */
+    _renderDoNotShowWarning() {
+        if (!this.props.doNotShowWarningLocalStorageKey
+            || !this.props.doNotShowWarningTextKey) {
+            return null;
+        }
+
+        return (
+            <h5>
+                <Label
+                    label
+                    = { this.props.t(this.props.doNotShowWarningTextKey) } />
+                <Toggle onChange = { this._onDontShowChanged } />
+            </h5>);
+    }
+
+    /**
+     * Updates state when toggle state changes.
+     *
+     * @param {Object} e - The event coming from Toggle component.
+     * @private
+     * @returns {void}
+     */
+    _onDontShowChanged(e) {
+        this.setState({
+            doNotShowWarningChecked: e.target.checked
```

Comment:
You should probably declare a default state to be consistent with other components that use state. Actually, you might be able to just do that in the constructor because you have access to props.